core/src/test/java/com/radixdlt/rev2/LedgerProofsGcTest.java

```diff
@@ -0,0 +1,200 @@
+        .addPhysicalNodes(PhysicalNodeConfig.createBatch(1, true))
+        .messageSelector(firstSelector())
+        .messageMutator(MessageMutator.dropTimeouts())
+        .functionalNodeModule(
+            new FunctionalRadixNodeModule(
+                NodeStorageConfig.tempFolder(folder),
+                true,
+                SafetyRecoveryConfig.MOCKED,
+                ConsensusConfig.of(1000),
+                LedgerConfig.stateComputerNoSync(
+                    new StateComputerConfig.REv2StateComputerConfig(
+                        Network.INTEGRATIONTESTNET.getId(),
+                        GenesisBuilder.createTestGenesisWithNumValidators(
+                            1,
+                            Decimal.ONE,
+                            GenesisConsensusManagerConfig.Builder.testWithRoundsPerEpoch(
+                                epochLength)),
+                        new DatabaseFlags(false, false),
+                        REV2ProposerConfig.transactionGenerator(
+                            new MbTransactionGenerator(NetworkDefinition.INT_TEST_NET), 3),
+                        false,
+                        StateHashTreeGcConfig.forTesting(),
+                        new LedgerProofsGcConfig(
+                            UInt32.fromNonNegativeInt(GC_INTERVAL_SEC),
+                            UInt64.fromNonNegativeLong(mostRecentFullResolutionEpochCount)),
+                        false))));
+  }
+
+  @Test
+  public void node_prunes_non_critical_proofs_from_configured_old_enough_epochs() {
+    // Arrange: configure ledger proof gc to keep 2 most recent completed epochs not pruned
+    var mostRecentFullResolutionEpochCount = 2;
+    /// ... 6 rounds * 3 transactions * 1 MB = 18MB; Limit is 12 MB - so we require 2 proofs
+    var epochLength = 6;
+    try (var test = createTest(mostRecentFullResolutionEpochCount, epochLength)) {
+      test.startAllNodes();
+
+      // Act: advance to epoch 5
+      test.runUntilState(NodesPredicate.nodeAt(0, NodePredicate.atOrOverEpoch(5)), 10000);
+
+      // Assert: after an async GC, we expect certain number of proofs in each epoch:
+      Awaitility.await()
+          .atMost(2 * GC_INTERVAL_SEC, TimeUnit.SECONDS)
+          .untilAsserted(
+              () -> {
+                var stateReader = test.getInstance(0, TestStateReader.class);
+                // - The epoch 5 has just started, hence it has no proofs
+                assertThat(stateReader.countProofsWithinEpoch(5)).isEqualTo(0);
+                // - The 2 most recent completed epochs contain all their proofs (i.e. not pruned)
+                assertThat(stateReader.countProofsWithinEpoch(4)).isEqualTo(epochLength);
+                assertThat(stateReader.countProofsWithinEpoch(3)).isEqualTo(epochLength);
+                // - This epoch was pruned, and it has 2 proofs (1 would not fit the 12MB limit)
+                assertThat(stateReader.countProofsWithinEpoch(2)).isEqualTo(2);
```

Comment:
Hmm okay maybe my bug spot was wrong in the gc code, but I'll leave it to you to work out what I wrongly observed.